src/network/network.cpp

```diff
@@ -847,21 +885,25 @@ void Network::UpdateClient()
 				status = NETWORK_STATUS_CONNECTED;
 				server_connection.ResetLastPacketTime();
 				Client_Send_AUTH(gConfigNetwork.player_name, "");
-				window_network_status_open("Authenticating...");
+				char str_authenticating[256];
+				format_string(str_authenticating, STR_MULTIPLAYER_AUTHENTICATING, NULL);
+				window_network_status_open(str_authenticating);
 			}
 		}
 	}break;
 	case NETWORK_STATUS_CONNECTED:
 		if (!ProcessConnection(server_connection)) {
-			std::string errormsg = "Disconnected";
-			if (server_connection.last_disconnect_reason) {
-				errormsg += ": ";
-				errormsg += server_connection.last_disconnect_reason;
-			}
 			if (server_connection.authstatus == NETWORK_AUTH_REQUIREPASSWORD) { // Do not show disconnect message window when password window closed/canceled
 				window_network_status_close();
 			} else {
-				window_network_status_open(errormsg.c_str());
+				char str_disconnected[256], str_conditional_buffer[256];
+				str_conditional_buffer[0] = '\0';
+				if (server_connection.getLastDisconnectReason()) {
```

Comment:
Yeah, that's what I meant. Not sure if any localisation will actually need
it, but better be safe than sorry.
Op vr 19 feb. 2016 om 16:53 schreef Martin Černáč <notifications@github.com>

> In src/network/network.cpp
> <https://github.com/OpenRCT2/OpenRCT2/pull/2968#discussion_r53477563>:
>
> >  			if (server_connection.authstatus == NETWORK_AUTH_REQUIREPASSWORD) { // Do not show disconnect message window when password window closed/canceled
> >  				window_network_status_close();
> >  			} else {
> > -				window_network_status_open(errormsg.c_str());
> > +				char str_disconnected[256], str_conditional_buffer[256];
> > +				str_conditional_buffer[0] = '\0';
> > +				if (server_connection.getLastDisconnectReason()) {
>
> I'm not sure I understand correctly - are you suggesting to have two
> translatable strings to avoid the conditional string("Disconnected" and
> "Disconnected: {STRING}")? If that's the case, I agree, will do.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenRCT2/OpenRCT2/pull/2968/files#r53477563>.
>